src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/__tests__/components/base/ButtonComponent.test.tsx

```diff
@@ -61,6 +62,7 @@ describe('components/base/ButtonComponent.tsx', () => {
           disabled={mockDisabled}
           formDataCount={formDataCount}
           language={mockLanguage}
+          {...({} as IComponentProps)}
```

Comment:
Why does this need to be added?

Reply:
Because `IComponentProps` does not have all the missing fields optional. They will always be provided in proper rendering in generic component, so non nullable seems correct. When testing, it does not make sense to send unused properties, so this seemed like the right compromise.

Reply:
The thing is that I have added the same "base" contract for all components (so that it is easy to see what is available with intellisense when writing or modifying a component). Not all components use the full set of available props, and sometimes  the tests don't exercise the functionality that needs the props.

I think this is a fine solution for test code. The alternative would probably be some sort of `{...defaultComponentProps}` at the top with (non)sensical do nothing mocks for functionality that is not used/tested. I'd rather prefere the tests fail with `can not access xxx on undefined`.